packages/react-components/react-progress/src/components/Progress/useProgressState.tsx

```diff
@@ -0,0 +1,35 @@
+import type { ProgressState, ProgressProps } from './Progress.types';
+import { progressCssVars } from './useProgressStyles';
+
+// if the percentComplete is near 0, don't animate it.
+// This prevents animations on reset to 0 scenarios
+const ZERO_THRESHOLD = 0.01;
```

Comment:
You could have a usePrevious hook and then compare the new and old values to determine if you should animate.